lib/vagrant-vcloud/action/forward_ports.rb

```diff
@@ -64,24 +70,26 @@ def forward_ports
           if !ports.empty?
             # We only need to forward ports if there are any to forward
             @logger.debug("Port object to be passed: #{ports.inspect}")
-            @logger.debug("Current network id #{cfg.vdc_network_id}")
 
             ### Here we apply the nat_rules to the vApp we just built
-            add_ports = cnx.add_vapp_port_forwarding_rules(
-              vapp_id,
-              'Vagrant-vApp-Net',
-              {
-                :fence_mode       => 'natRouted',
-                :parent_network   => cfg.vdc_network_id,
-                :nat_policy_type  => 'allowTraffic',
-                :nat_rules        => ports
-              }
-            )
-
-            wait = cnx.wait_task_completion(add_ports)
-
-            if !wait[:errormsg].nil?
-              raise Errors::ComposeVAppError, :message => wait[:errormsg]
+            ports.values.each do |port|
```

Comment:
@marc-ta Changing line 108 from `ports.each` to `ports.values.each` seems to fix the problem with the edge gateway. 

Could you please update this and rebase? :heart: